### PR TITLE
Validate recipient object

### LINKED ISSUE #302: sendTransaction: Recipient object must be validated
If you try to use `window.marina.sendTransaction` with a non compliant Recipient object without `asset` it will not block you and let proceed till coin selection.

```
await marina.sendTransaction([{address: "tex1qhdvpxjjjumvms8qhdwfcmjg27ww0xr5nsf406f", value: 5000000}]); 
```

It should throw error early.

We should check:

1) address & value being present ALWAYS.
2) If asset is not present either default to L-BTC of the selected network since is known 
3) address must match and be valid for the current network.

### PR DESCRIPTION
**Pull request by bordalix**
Fixes #302

This PR:
- validates asset and value on all recipients in sendTransaction()
- moves value validation from provider.ts to marinaBroker.ts

@tiero please review

### CODE REVIEW

File: src/content/marina/marinaBroker.ts [hunk 1]
```diff
@@ -178,10 +178,22 @@ export default class MarinaBroker extends Broker {
...
 
+          // validate object recipient (asset and value)
+          // if no asset is present, assume lbtc for the current network
+          for (const recipient of recipients) {
+            if (!recipient.asset) {
+              if (!lbtc) throw new Error('missing asset on recipient');
+              recipient.asset = lbtc;
+            }
+            if (!recipient.value) throw new Error('missing value on recipient');
+            if (recipient.value < 0) throw new Error('negative value on recipient');
```
Comment (tiero) [inline src/content/marina/marinaBroker.ts:194]: worth checking floating point numbers as not valid

**Review by tiero — APPROVED** [review]

File: src/content/marina/marinaBroker.ts (hunk 1 above)
Comment (bordalix) [reply src/content/marina/marinaBroker.ts]: Adding Number.isSafeInteger in 277de1f

But, `Number.isSafeInteger(1.0)` is true, which means a value of `1.0` would continue downstream.
Should I `recipient.value = parseInt(recipient.value, 10)` after this test?

Comment (tiero) [reply src/content/marina/marinaBroker.ts]: yes add extra parseInt indeed

Comment (tiero) [reply src/content/marina/marinaBroker.ts]: good catch

**Review by tiero — CHANGES_REQUESTED** [review]
parseInt after isSafeInteger

**Pull request merged**